testing/test_cache.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+
+pytest_plugins = "pytester",
+
+class TestNewAPI:
+    def test_config_cache_makedir(self, testdir):
+        testdir.makeini("[pytest]")
+        config = testdir.parseconfigure()
+        with pytest.raises(ValueError):
+            config.cache.makedir("key/name")
+
+        p = config.cache.makedir("name")
+        assert p.check()
+
+    def test_config_cache_dataerror(self, testdir):
+        testdir.makeini("[pytest]")
+        config = testdir.parseconfigure()
+        cache = config.cache
+        pytest.raises(TypeError, lambda: cache.set("key/name", cache))
+        config.cache.set("key/name", 0)
+        config.cache._getvaluepath("key/name").write("123invalid")
+        val = config.cache.get("key/name", -2)
+        assert val == -2
+
+    def test_config_cache(self, testdir):
+        testdir.makeconftest("""
+            def pytest_configure(config):
+                # see that we get cache information early on
+                assert hasattr(config, "cache")
+        """)
+        testdir.makepyfile("""
+            def test_session(pytestconfig):
+                assert hasattr(pytestconfig, "cache")
+        """)
+        result = testdir.runpytest()
+        assert result.ret == 0
+        result.stdout.fnmatch_lines(["*1 passed*"])
+
+    def XXX_test_cachefuncarg(self, testdir):
```

Comment:
I figured. :smile:

(btw, did you notice the comments I made about the docs? Just making sure)